Game/MyProject/Game/Helper/HelperManager.cpp

```diff
@@ -0,0 +1,553 @@
+        
+        indices.push_back(3);
+        indices.push_back(2);
+        indices.push_back(6);
+        
+        indices.push_back(3);
+        indices.push_back(6);
+        indices.push_back(8);
+        
+        indices.push_back(3);
+        indices.push_back(8);
+        indices.push_back(9);
+        
+        
+        indices.push_back(4);
+        indices.push_back(9);
+        indices.push_back(5);
+        
+        indices.push_back(2);
+        indices.push_back(4);
+        indices.push_back(11);
+        
+        indices.push_back(6);
+        indices.push_back(2);
+        indices.push_back(10);
+        
+        indices.push_back(8);
+        indices.push_back(6);
+        indices.push_back(7);
+        
+        indices.push_back(9);
+        indices.push_back(8);
+        indices.push_back(1);
+        
+        indices.push_back(10);
+        indices.push_back(11);
+        indices.push_back(5);
+        
+        
+        RefineTriangles(vertices,indices,4);
+        
+        sphereMesh->setVertexBuffer(vertices, format);
+        sphereMesh->setIndexBuffer(indices);
+        
+        return sphereMesh;
+    }
+    //---------------------------------------------------------------------------
+    // @HelperManager::CreateMeshInstance()
+    //---------------------------------------------------------------------------
+    std::shared_ptr<MeshInstance> CreateMeshInstance(const std::shared_ptr<Mesh> &mesh,
+                                                     const char* shaderName)
+    {
+        std::shared_ptr<MeshInstance> meshInstance=std::make_shared<MeshInstance>();
+        meshInstance->setMesh(mesh);
+        meshInstance->setShader(shaderName);
+        
+        return meshInstance;
+    }
+    
+    //---------------------------------------------------------------------------
+    // @HelperManager::CreateMeshInstance()
+    //---------------------------------------------------------------------------
+    std::shared_ptr<MeshInstance> CreateMeshInstance(const std::shared_ptr<Mesh> &mesh,
+                                                     const std::vector<std::string>& uniforms,
+                                                     const char* shaderName)
+    {
+        std::shared_ptr<MeshInstance> meshInstance=std::make_shared<MeshInstance>();
+        meshInstance->setMesh(mesh);
+        meshInstance->setShader(shaderName);
+        meshInstance->addShaderUniforms(uniforms);
+        
+        return meshInstance;
+    }
+    //---------------------------------------------------------------------------
+    // @HelperManager::BuildSphere()
+    //---------------------------------------------------------------------------
+    std::shared_ptr<SceneNode>
+    BuildSphere(const std::shared_ptr<MeshInstance>&meshInstance,bool wireframeValue)
+    {
+        RenderPacket myRenderPacket;
+        myRenderPacket._use_blend = true;
+        myRenderPacket._use_depth = true;
```

Comment:
these are generic primitive builder methods.. are you sure blend and depth testing will be enabled for all their uses?